nodejs/week2/Mealsharing/src/backend/routes/meal.js

```diff
@@ -0,0 +1,80 @@
+const meals = require("../data/meals.json");
+const express = require("express");
+const router = express.Router();
+router.get("/", (req, res) => {
+          // default route to access the all meals
+          if (!meals) {
+                    res.status(404).send("Data does not exists");
+          }
+          res.send(meals);
+});
+router.get("/meals/:id", (req, res) => {
+          // route with id
+          const getId = parseInt(req.params.id);
+          const getMeal = meals.filter((meal) => meal.id === getId);
+          if (!getMeal.length) {
+                    res.status(404).send("Meal does not  exists of this id");
+          }
+          res.send(getMeal);
+});
+
+router.get("/meals", (req, res) => {
+          // Query method
+          const keyValue = Object.keys(req.query);
```

Comment:
This approach is not the best, because you can filter array of meals by only one parameter, which goes first in req.query.
You can use destructuring here
`const {maxPrice, title, createdAfter, limit} = req.query`
in that case you don't need to check in your conditions if it is equal to keyValue[0], just check if it exists.